src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -364,13 +364,30 @@ public void DefComplexTypeWithFunc()
                 AllowsSideEffects = false,
             };
 #pragma warning disable CS0618 // Type or member is obsolete
-            recalcEngine.DefineType("Complex = Type({ A: {B: Number}});", parserOptions);
+            Assert.False(recalcEngine.DefineType("Complex = Type({ A: {B: Number}});", parserOptions).Any());
 #pragma warning restore CS0618 // Type or member is obsolete
-            IEnumerable<ExpressionError> enumberable = recalcEngine.DefineFunctions("foo(p: Complex): Number = p.A.B;").Errors;
-            Assert.False(enumberable.Any());
+            IEnumerable<ExpressionError> enumerable = recalcEngine.DefineFunctions("foo(p: Complex): Number = p.A.B;").Errors;
+            Assert.False(enumerable.Any());
             Assert.Equal(1.0, recalcEngine.Eval("foo({A: {B: Float(1.0)}})").ToObject());
         }
 
+        [Fact]
+        public void DefTableType()
+        {
+            var config = new PowerFxConfig();
+            var recalcEngine = new RecalcEngine(config);
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = false,
+            };
+#pragma warning disable CS0618
+            Assert.False(recalcEngine.DefineType("People = Type([{Age: Number}]);", parserOptions).Any());

```

Comment:
> Type([{Age: Number}] [](http://example.com/codeflow?start=58&length=21)

Does it work with complex type?

For example,
Complex = Type({ A: Number });
 Type([{Age: Complex}])

Reply:
Yes